### PR TITLE
Task sf/j294 add favourite episodes functionality

### PR DESCRIPTION
**Pull request by Sayonarko**
favour

### CODE REVIEW

File: backend/src/api/episodes/episodes.api.ts [hunk 1]
```diff
@@ -54,6 +59,32 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
     }),
   );
 
+  episodeRouter.get(
+    EpisodesApiPath.FAVOURITES_$ID,
```
Comment (what1s1ove) [inline backend/src/api/episodes/episodes.api.ts]:
```suggestion
    EpisodesApiPath.FAVOURITES_$USER_ID,
```

File: frontend/src/store/user-favourite-episodes/actions.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AsyncThunkConfig, UserFavouriteEpisodeResponse, LoadFavouriteEpisodesPayload } from 'common/types/types';
+import { ActionType } from './common';
+
+const loadFavouriteEpisodes = createAsyncThunk<UserFavouriteEpisodeResponse, LoadFavouriteEpisodesPayload, AsyncThunkConfig>(
+  ActionType.LOAD_FAVOURITE_EPISODES,
+  async (payload, { extra }) => {
+    const { episodeApi } = extra;
+
+    return await episodeApi.getFavoriteEpisodesByUserId(payload);
```
Comment (what1s1ove) [inline frontend/src/store/user-favourite-episodes/actions.ts]:
```suggestion
    return episodeApi.getFavoriteEpisodesByUserId(payload);
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]
add podcast name to the table
move episode-table back, because we will have a different tables

**Review by JuliaDzyuba — DISMISSED** [review]

File: backend/src/api/episodes/episodes.api.ts [hunk 3]
```diff
@@ -77,6 +119,16 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
     }),
   );
 
+  episodeRouter.delete(
+    EpisodesApiPath.FAVOURITES,
+    checkAuthMiddleware(HttpMethod.DELETE),
+    handleAsyncApi(async (req, res) => {
```
Comment (what1s1ove) [inline backend/src/api/episodes/episodes.api.ts:135]: always add validation for requests which use body

File: frontend/src/components/user-favourite-episodes/components/favourite-episode-table/helpers/get-formatted-date.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { DateFormatType } from 'common/enums/enums';
```
Comment (what1s1ove) [inline frontend/src/components/user-favourite-episodes/components/favourite-episode-table/helpers/get-formatted-date.ts]: `frontend/src/components/user-favourite-episodes/components/favourite-episode-table/helpers/get-formatted-date/get-formatted-date.helper.ts`

File: shared/src/common/types/episode/episode.type.ts [hunk 5]
```diff
@@ -15,6 +15,7 @@ type Episode = {
   type: EpisodeType;
   shownotes: Shownote[];
   description: string;
+  podcast?: Podcast;
```
Comment (what1s1ove) [inline shared/src/common/types/episode/episode.type.ts]: we can't change the origin episode type

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — DISMISSED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**